addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -49,6 +50,20 @@ def _get_specific_rendering_values(self, processing_values):
         converted_amount = payment_utils.to_minor_currency_units(self.amount, self.currency_id)
         base_url = self.provider_id.get_base_url()
         return_url_params = {'reference': self.reference}
+
+        phone = self.partner_phone
+        if not self.partner_phone:
+            raise ValidationError("Razorpay: " + _("Phone number is required."))
+        else self.partner_phone:
+            # sanitize partner phone (+9196385 44240)
+            country_code = self.partner_country_id.code
+            country_phone_code = self.partner_country_id.phone_code
+            phone = phone_sanitize_numbers(
+                [self.partner_phone], country_code, country_phone_code
+            ).get(self.partner_phone, {}).get('sanitized')
+            if not phone:
+                raise ValidationError("Razorpay: " + _("Incorrect phone number."))
```

Comment:
I was expecting something like this instead:

```suggestion
        phone = self.partner_phone
        error_message = _("Partner's phone number is missing.")
        if self.partner_phone:
            # sanitize partner phone
            country_code = self.partner_country_id.code
            country_phone_code = self.partner_country_id.phone_code
            phone_info = phone_sanitize_numbers([self.partner_phone], country_code, country_phone_code)
            phone = phone_info.get(self.partner_phone, {}).get('sanitized')
            error_message = phone_info.get(self.partner_phone, {}).get('msg')
        if not phone:
            raise ValidationError("Razorpay: " + error_message)
```